spikeinterface/sortingcomponents/peak_pipeline.py

```diff
@@ -18,32 +18,34 @@
 from spikeinterface.core import get_channel_distances
 
 class PipelineNode:
-    """
-    This is a generic object that will make some computation on peak given a buffe of traces.
-    Typically used for exctrating features (amplitudes, localization, ...)
-    
-    A Node can optionally connect to other nodes with the parents and receive inputs from others.
-    """
-    def __init__(self, recording, return_output=True, parents=None, name=None):
+    def __init__(
+        self, recording: BaseRecording, return_output: bool = True, parents: Optional[List[Type["PipelineNode"]]] = None
+    ):
+        """
+        This is a generic object that will make some computation on peaks given a buffer of traces.
+        Typically used for exctrating features (amplitudes, localization, ...)
+
+        A Node can optionally connect to other nodes with the parents and receive inputs from them.
+
+        Parameters
+        ----------
+        recording : BaseRecording
+            The recording object.
+        parents : Optional[List[PipelineNode]], optional
+            Pass parents nodes to perform a previous computation, by default None
+        return_output : bool, optional
+            Whether or not the output of the node is returned by the pipeline, by default False
+        """
+
         self.recording = recording
         self.return_output = return_output
         if isinstance(parents, str):
             # only one parents is allowed
             parents = [parents]
         self.parents = parents
         
-        self._kwargs = dict(
-            return_output=return_output,
-        )
-        if parents is not None:
-            self._kwargs['parents'] = parents
+        self._kwargs = dict()
```

Comment:
I think I likde the idea that parent and return_output where handle in the base class.
So we never forget to push in then in all classes

Reply:
There are some children that are still acessing kwars. I will remove that in another refactor.